datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -17,30 +17,33 @@
 package quasar.physical.mongo
 
 import slamdata.Predef._
+import scala.util.Either
 
 import cats.effect._
-import cats.effect.concurrent.{MVar, MVar2}
-import cats.effect.syntax.bracket._
 import cats.implicits._
 
-import fs2.concurrent.{NoneTerminatedQueue, Queue}
 import fs2.Stream
 
 import quasar.api.resource.ResourcePath
 import quasar.connector.{MonadResourceErr, ResourceError}
+import quasar.physical.mongo.contrib.fs2.StreamSubscriber
 import quasar.physical.mongo.MongoResource.{Collection, Database}
 import quasar.{IdStatus, ScalarStages}
 
 import org.bson.{Document => _, _}
-import com.mongodb.{Block, ConnectionString}
-import com.mongodb.connection.{ClusterSettings, SslSettings}
+
 import org.mongodb.scala._
+import org.mongodb.scala.connection.{ClusterSettings, SslSettings}
+
+import com.mongodb.Block
+
+import java.lang.{ClassCastException, NumberFormatException}
 
 import shims.equalToCats
 
 final class Mongo[F[_]: MonadResourceErr: ConcurrentEffect: ContextShift] private[mongo](
     client: MongoClient,
-    batchSize: Long,
+    batchSize: Int,
```

Comment:
Has the unit changed here? If not, does it need to change to Int? Since I think a Long is useful here.

Reply:
It appears to have always been [bounded by `Int`](https://github.com/precog/quasar-datasource-mongo/pull/577/files/0f042e521e77dc89128b25784d796ace56cd4615#diff-3bf734f68f9be31725373754e2db7c2239318c635dd0a30d4080381c983639afL91) not sure why it used to be `Long` here.

Reply:
👍  Was just triggered by the diff